src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -407,6 +429,95 @@ public override async Task<bool> DrainAsync(int maxDrainBytes)
                     cts?.Dispose();
                 }
             }
+
+            private void ParseHeaderNameValue(ReadOnlySpan<byte> line, HttpResponseHeaders trailers)
+            {
+                Debug.Assert(line.Length > 0);
+
+                int pos = 0;
+                while (line[pos] != (byte)':' && line[pos] != (byte)' ')
+                {
+                    pos++;
+                    if (pos == line.Length)
+                    {
+                        // Invalid header line that doesn't contain ':'.
+                        return;
+                    }
+                }
+
+                if (pos == 0)
+                {
+                    // Invalid empty header name.
+                    return;
+                }
+
+                if (!HeaderDescriptor.TryGet(line.Slice(0, pos), out HeaderDescriptor descriptor))
+                {
+                    // Invalid header name
+                    ThrowInvalidHttpResponse();
+                }
+
+                if (descriptor.KnownHeader != null && disallowedHeaders.Contains(descriptor.KnownHeader))
+                {
+                    // Disallowed trailer fields.
+                    // A recipient MUST ignore fields that are forbidden to be sent in a trailer.
+                    return;
+                }
+
+                // Eat any trailing whitespace
+                while (line[pos] == (byte)' ')
+                {
+                    pos++;
+                    if (pos == line.Length)
+                    {
+                        // Invalid header line that doesn't contain ':'.
+                        return;
+                    }
+                }
+
+                if (line[pos++] != ':')
+                {
+                    // Invalid header line that doesn't contain ':'.
+                    return;
+                }
+
+                // Skip whitespace after colon
+                while (pos < line.Length && (line[pos] == (byte)' ' || line[pos] == (byte)'\t'))
+                {
+                    pos++;
+                }
+
+                string headerValue = descriptor.GetHeaderValue(line.Slice(pos));
+
+                // Ignore the return value from TryAddWithoutValidation,
+                // if the header can't be added, we silently drop it.
+                trailers.TryAddWithoutValidation(descriptor, headerValue);
+            }
+
+            private static readonly List<KnownHeader> disallowedHeaders = new List<KnownHeader>
```

Comment:
A few things:
- Nit: s_disallowedTrailingHeaders
- This should be at the top of the file.
- Seems like this should be a HashSet to allow for faster O(1) lookup?

Reply:
Will change.